irods/configuration/__init__.py

```diff
@@ -0,0 +1,17 @@
+class iRODSConfiguration(object):
+    __slots__ = ()
+
+def getter(category, setting):
+    return lambda:getattr(globals()[category], setting)
+
+# classes for building configuration categories
+
+class DataObjects(iRODSConfiguration):
+    __slots__ = ('auto_close',)
+    def __init__(self):
+        self.auto_close = False
+
+
+# configuration categories
+
+data_objects = DataObjects()
```

Comment:
This is what allows the following to work:
```
import irods.configuration
print ('default value of auto-close is:',irods.configuration.data_objects.auto_close)

# after this statement all data objects opened will be "managed", ie automatically closed:
irods.configuration.data_objects.auto_close = True
# this would generate an error due to the __slots__ definition restricting valid attribute names:
irods.configuration.data_objects.auto_close_enable = True
```

Reply:
Wondering if this should be renamed irods.client_configuration. @korydraughn @alanking @trel ?
("irods.configuration" could be misleading.)
One can always abbreviate: `import irods.client_configuration as client_config`

Reply:
> Wondering if this should be renamed irods.client_configuration. @korydraughn @alanking @trel ? ("irods.configuration" could be misleading.) One can always abbreviate: `import irods.client_configuration as client_config`

I agree. `client_configuration` is clearer.